packages/grid/x-data-grid/src/internals/components/containers/GridRootStyles.ts

```diff
@@ -73,6 +73,10 @@ export const GridRootStyles = styled('div', {
     [`&.${gridClasses.autoHeight}`]: {
       height: 'auto',
     },
+    [`& .${gridClasses['virtualScrollerContent--overflowed']} .${gridClasses['row--lastVisible']} .${gridClasses.cell}`]:
```

Comment:
The content is not supposed to overflow when `autoHeight` is on. We need to also remove the border in this condition. In https://deploy-preview-3519--material-ui-x.netlify.app/components/data-grid/layout/#auto-height sometimes I still see the double border.

![image](https://user-images.githubusercontent.com/42154031/155384737-78249354-d0cc-4212-8d0a-0455b8145ca6.png)

Reply:
Ah yes thanks for the catch!